src/inc/sift-events/sift-events.php

```diff
@@ -715,13 +796,40 @@ public static function transaction( \WC_Order $order, string $status, string $tr
 	 * @return void
 	 */
 	public static function change_order_status( string $order_id, \WC_Order $order, array $status_transition ) {
-
 		if ( ! Sift_Event_Types::can_event_be_sent( Sift_Event_Types::$order_status ) ) {
 			return;
 		}
 
+		// Check if this is a free order and if we should send a create_order event too.
+		$is_free_order = self::is_free_order( $order );
+
+		// For free orders that are just being created (moving to pending/processing), send the create_order event.
+		if ( $is_free_order &&
+			in_array( $status_transition['to'], array( 'pending', 'processing' ), true ) &&
+			Sift_Event_Types::can_event_be_sent( Sift_Event_Types::$create_order ) ) {
+			if ( function_exists( 'wc_get_logger' ) ) {
+				wc_get_logger()->debug(
+					sprintf( 'Free order detected at status change - triggering create_order event for order %s', $order_id ),
+					array( 'source' => 'sift-free-orders' )
+				);
+			}
+
+			// This will trigger the create_order event for this free order.
+			self::create_order( $order_id, $order );
+		}
+
+		// Determine user and session context.
+		$user_id    = wp_get_current_user()->ID ?? null; // Check first for logged-in user.
```

Comment:
Getting the user from the $order might be better here as well

Reply:
So, with WCCOM's current setup, using just `'$user_id' => self::format_user_id( $order->get_user_id() ),` would work. However, in WooCommerce > Settings > Accounts & Privacy, if guest checkout is enabled, this will fail since WooCommerce wouldn't be creating a new user if they aren't already logged in.

We can go with the original, knowing that it will just fail on standard Woo sites, but we probably need to have a small discussion on how to better address this so it covers "guest checkout allowed" and "non guest checkout allowed" situations. I mentioned the sidecar plugin previously, but this problem would extend beyond WCCOM; any Woo site could have the option configured either way.